techlibs/xilinx/abc_xc7.box

```diff
@@ -59,3 +59,1107 @@ $__ABC_LUT6 2000 0 7 1
 # Outputs: DPO SPO
 $__ABC_LUT7 2001 0 8 1
 0 1047 1036 877 812 643 532 478
+
+# Boxes used to represent the comb/seq behaviour of DSP48E1
+#   With abc_map.v responsible for disconnecting inputs to
+#   the combinatorial DSP48E1 model by a register (e.g.
+#   disconnecting A when AREG, MREG or PREG is enabled)
+#   this mux captures the existence of a replacement path
+#   between AREG/BREG/CREG/etc. and P/PCOUT.
+#   Since the Aq/ADq/Bq/etc. inputs are assumed to arrive at
+#   the mux at zero time, the combinatorial delay through
+#   these muxes thus represents the clock-to-q delay at
+#   P/PCOUT.
+$__ABC_DSP48E1_MULT_P_MUX 2100 0 103 48
+# A AD    B    C D                                                                                               I    M P                                                                                                Pq
+2952 - 2813 1687 - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 1671 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 329
```

Comment:
What about a Python script to generate this massive array of numbers (maybe I have missed one?)

Reply:
Right now, this information is imported from a spreadsheet by hand. The vision is to generate box/lut files automatically, but until that day, I don't believe there's a lot of value in writing and maintaining a script to generate this information since it is expected to be constant once finalised. If someone else wants to do one and maintain it, I'd be happy to help.